pkg/storkctlcli/storkctlcli.go

```diff
@@ -0,0 +1,161 @@
+				migCmdArgs = append(migCmdArgs, value)
+			}
+		}
+	}
+	cmd.SetArgs(migCmdArgs)
+	// execute the command
+	logrus.Infof("The storkctl command being executed is %v", migCmdArgs)
+	if err := cmd.Execute(); err != nil {
+		if err != nil {
+			return fmt.Errorf("Error in executing create migration schedule command: %v", err)
+		}
+	}
+	// Get the captured output as a string
+	actualOutput := outputBuffer.String()
+	logrus.Infof("Actual output is: %s", actualOutput)
+	return nil
+}
+
+func PerformFailoverOrFailbackStlCli(action, namespace, migSchdRef string, skipSourceOp bool, extraArgs map[string]string) (error, string) {
+	var pfCmdArgs []string
+	if namespace != "" {
+		migSchedNs = namespace 
+	}
+
+	factory := storkctl.NewFactory()
+	var outputBuffer bytes.Buffer
+	cmd := storkctl.NewCommand(factory, os.Stdin, &outputBuffer, os.Stderr)
+	if action == "failover" {
+		if skipSourceOp {
+			pfCmdArgs = []string{"perform", "failover", "--migration-reference", migSchdRef, "--namespace", migSchedNs, "--skip-source-operations"}
+		} else {
+			pfCmdArgs = []string{"perform", "failover", "--migration-reference", migSchdRef, "--namespace", migSchedNs}
+		}
+	} else {
+		pfCmdArgs = []string{"perform", "failback", "--migration-reference", migSchdRef, "--namespace", migSchedNs}
+	}
+	if extraArgs != nil {
+		for key, value := range extraArgs {
+			pfCmdArgs = append(pfCmdArgs, "--"+key)
+			if value != "" {
+				pfCmdArgs = append(pfCmdArgs, value)
+			}
+		}
+	}
+	cmd.SetArgs(pfCmdArgs)
+	// execute the command
+	logrus.Infof("The storkctl command being executed is %v", pfCmdArgs)
+	if err := cmd.Execute(); err != nil {
+		if err != nil {
+			return fmt.Errorf("Error in executing perform %v command: %v", action, err), ""
+		}
+	}
+	// Get the captured output as a string
+	actualOutput := outputBuffer.String()
+	logrus.Infof("Actual output is: %s", actualOutput)
+	return nil, actualOutput
+}
+
+func GetDRActionStatus(actionType storkv1.ActionType, actionName string, actionNamespace string, configPath string) (string, string, error) {
+	factory := storkctl.NewFactory()
+	var outputBuffer bytes.Buffer
+	cmd := storkctl.NewCommand(factory, os.Stdin, &outputBuffer, os.Stderr)
+	cmdArgs := []string{"get", string(actionType), actionName, "-n", actionNamespace, "--kubeconfig", configPath}
+	cmd.SetArgs(cmdArgs)
+	if err := cmd.Execute(); err != nil {
+		if err != nil {
+			return "", "", fmt.Errorf("Error in executing perform %v command: %v", string(actionType), err)
+		}
+	}
+	// Get the captured output as a string
+	actualOutput := outputBuffer.String()
+	log.InfoD("Actual output is: %s", actualOutput)
+	output := strings.Split(actualOutput, "\n")
+	startIndex := make(map[string]int)
+	columns := []string{"NAME", "CREATED", "STAGE", "STATUS", "MORE INFO"}
+	for _, column := range columns {
+		startIndex[column] = strings.Index(output[0], column)
+	}
+	currentStatus := strings.TrimSpace(output[1][startIndex["STATUS"]:startIndex["MORE INFO"]])
```

Comment:
Instead of parsing columns can we get a `-j` output and then use unmarshal json? If the output format changes tomorrow this code might fail.

Reply:
Used GetAction method and it worked.